File: GestorMaestro/GestorMaestro.Web/Modules/ServerTypes/Escuela/MateriasColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MateriasRow } from "./MateriasRow";

export interface MateriasColumns {
    IdMateria: Column<MateriasRow>;
    NombreMateria: Column<MateriasRow>;
}

export class MateriasColumns extends ColumnsBase<MateriasRow> {
    static readonly columnsKey = 'Escuela.Materias';
    static readonly Fields = fieldsProxy<MateriasColumns>();
}

File: GestorMaestro/GestorMaestro.Web/Modules/Escuela/Materias/MateriasColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace GestorMaestro.Escuela.Columns;

[ColumnsScript("Escuela.Materias")]
[BasedOnRow(typeof(MateriasRow), CheckNames = true)]
public class MateriasColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int IdMateria { get; set; }
    [EditLink]
    public string NombreMateria { get; set; }
}